src/background/install-data-generator.ts

```diff
@@ -1,25 +1,23 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
-import { BrowserAdapter } from './browser-adapter';
+import { StorageAPI } from './browser-adapters/storage-adapter';
 import { InstallationData } from './installation-data';
 import { LocalStorageDataKeys } from './local-storage-data-keys';
 
 export class InstallDataGenerator {
     private generateGuid: () => string;
     private dateGetter: () => Date;
     private installationData: InstallationData;
-    private browserAdapter: BrowserAdapter;
 
     constructor(
         initialInstallationData: InstallationData,
         generateGuid: () => string,
         dateGetter: () => Date,
-        browserAdapter: BrowserAdapter,
+        private readonly storageAdapter: StorageAPI,
```

Comment:
For consistency, should we also change the other parameters in the constructor to look the same as the one for `storageAdapter`?

Reply:
all except `initialInstallationData`, which is use to initialize an internal, private property which is not `readonly`.